server/services/preview-button.js

```diff
@@ -36,7 +36,8 @@ module.exports = ( { strapi } ) => ( {
     }
 
     draftUrl = `${draftUrl}?${qs.stringify( draftParams )}`;
-    publishedUrl = `${publishedUrl}/${entity[ targetField ]}`;
+    // Remove entity targetField if is a SingleType
+    publishedUrl = entity[ targetField ] ? `${publishedUrl}/${entity[ targetField ]}` : `${publishedUrl}`;
```

Comment:
Why remove the `targetField` here? This would render the `publishedUrl` useless because it would not have `slug` path for the entity.

Reply:
because if it is a single entity it does not have an incremental id, or unique slug for, it can be differential for the path. For example if I have "about page" i want to show in preview only "/about" url.